### ISSUE TITLE
Membership fees

### ISSUE THREAD

**Issue by djulien19** [OCA/vertical-community#7]
Right now in Odoo we could have a beginning date and an end date for a membership. But in associations, they use more often a period of time: a month, a quarter, a semester, a year.
And it has been done by Heureux Cyclage on their project inside every memberpage. I still would like to get membership fees with this period of time as articles.

**Comment by lucode** [comment]
Could you specify more detailed please? What is the difference? It is only a bit more work to set the range manually. Another thing would be to have fix membership periods as we have often here in Switzerland. A year for instance from Jan. to Dec.  cheers lucas

**Comment by djulien19** [comment]
Lucas,
If you subscribe today for this year in an association, you will pay the entire price for only one month. 
It would be great to get a range so as to pay for a period of time subscription. These ranges could be defined by the admin with number of days.
This is the work made by Heureux Cyclage:
http://git.heureux-cyclage.org/?p=burette/remembership.git;a=commitdiff;h=435ec7d1f0a1d180e9f76512b64c43f61ff5e295

**Comment by lucode** [comment]
Julien,
Now I see what you mean, but the other possibility is also highly welcome. The reason is very simple the associations and the members are familiar with this. And it is simple and straightforward. Every of my  memberships in Swiss associations is handled as this. 
But the problem with entrance and exits does exist. In manual system they normally don't charge the current year if you join the asoc. in the last three months of the year. Then you pay already for the upcoming year. In a electronic system this could maybe introduced too.

**Comment by djulien19** [comment]
Absolutely, we should keep both systems for associations. The time-based pricing is more and more used with the monthly or yearly basis. It will be also very interesting for start-ups that would like to sell hosting and services in Odoo, or even Odoo itself...

**Comment by YannickB** [comment]
I agree, this feature is highly welcome.
